packages/ui/components/data-table/filters/utils.ts

```diff
@@ -15,3 +18,111 @@ export const filtersSearchParams = {
 export function useFiltersSearchState() {
   return useQueryStates(filtersSearchParams);
 }
+
+export type FiltersSearchState = ReturnType<typeof useFiltersSearchState>[0];
+export type SetFiltersSearchState = ReturnType<typeof useFiltersSearchState>[1];
+export type ActiveFilter = NonNullable<FiltersSearchState["activeFilters"]>[number];
+
+export const textFilter = (cellValue: string, filterValue: TextFilterValue) => {
+  switch (filterValue.data.operator) {
+    case "equals":
+      return cellValue.toLowerCase() === (filterValue.data.operand || "").toLowerCase();
+    case "notEquals":
+      return cellValue.toLowerCase() !== (filterValue.data.operand || "").toLowerCase();
+    case "contains":
+      return cellValue.toLowerCase().includes((filterValue.data.operand || "").toLowerCase());
+    case "notContains":
+      return !cellValue.toLowerCase().includes((filterValue.data.operand || "").toLowerCase());
+    case "startsWith":
+      return cellValue.toLowerCase().startsWith((filterValue.data.operand || "").toLowerCase());
+    case "endsWith":
+      return cellValue.toLowerCase().endsWith((filterValue.data.operand || "").toLowerCase());
+    case "isEmpty":
+      return cellValue.trim() === "";
+    case "isNotEmpty":
+      return cellValue.trim() !== "";
+    default:
+      return false;
+  }
+};
+
+export const isTextFilterValue = (filterValue: unknown): filterValue is TextFilterValue => {
+  return (
+    typeof filterValue === "object" &&
+    filterValue !== null &&
+    "type" in filterValue &&
+    filterValue.type === "text"
+  );
+};
+
+export const isSelectFilterValue = (filterValue: unknown): filterValue is SelectFilterValue => {
+  return Array.isArray(filterValue) && filterValue.every((item) => typeof item === "string");
+};
+
+export function makeWhereClause(columnName: string, filterValue: FilterValue) {
```

Comment:
This is a utility function for prisma for the backend to build proper where clause.